### PR TITLE
Adding and updating release-specific docs

### PR DESCRIPTION
**Pull request by bridgetkromhout**
Notes on how to add and (if necessary) change any release-specific docs. @flynnduism for any clarifications, corrections, additions needed.

Signed-off-by: Bridget Kromhout <bridget@kromhout.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +32,15 @@ type: docs
 * inclusion of `type: docs` is important for the theme to properly index the site contents
 * the `linkTitle` attribute allows you to simplify the name as it appears in the left-side nav bar - ideally it should be short and clear - whereas the title can handle longform names for pages/documents.
 
+## Adding release-specific docs
+
+Steps to add a release-specific version of the docs site:
+
+1. Create a release branch from main, named after the released major and minor version, like [release-v0.8](https://github.com/openservicemesh/osm-docs/tree/release-v0.8).
```
Comment (flynnduism) [inline README.md:39]: In addition to creating the branch `release-vX.Y`, the branch should also be pushed `upstream` to this repo in order for it to be picked up by Netlify. 

By default, Netlify will deploy the branch to a url like https://release-v0-8--osm-docs.netlify.app/ which can be used to preview and test that it builds correctly. Though that testing could be carried out under the third bullet point

**Review by flynnduism — CHANGES_REQUESTED** [review]
Added a small note, would be good to instruct the pushing of the new branch once it's created.

Other than that this looks good to go. Thank you!

File: README.md (hunk 1 above)
Comment (bridgetkromhout) [reply README.md]: Thanks. I've edited and hopefully clarified; please check again for accuracy? Thanks.

**Review by flynnduism — APPROVED** [review]
/lgtm thanks!

**Pull request merged**